test/Microsoft.DotNet.ExecutablePackageObtainer.Tests/ExecutablePackageObtainerTests.cs

```diff
@@ -0,0 +1,201 @@
+        {
+            FilePath nugetConfigPath = WriteNugetConfigFileToPointToTheFeed();
+            var toolsPath = Path.Combine(Directory.GetCurrentDirectory(), Path.GetRandomFileName());
+
+            ExecutablePackageObtainer packageObtainer =
+                ConstructDefaultPackageObtainer(toolsPath);
+            ToolConfigurationAndExecutableDirectory toolConfigurationAndExecutableDirectory = packageObtainer.ObtainAndReturnExecutablePath(
+                packageId: TestPackageId,
+                packageVersion: TestPackageVersion,
+                nugetconfig: nugetConfigPath,
+                targetframework: _testTargetframework);
+
+            var executable = toolConfigurationAndExecutableDirectory
+                .ExecutableDirectory
+                .WithFile(
+                    toolConfigurationAndExecutableDirectory
+                        .Configuration
+                        .ToolAssemblyEntryPoint);
+
+            File.Exists(executable.Value)
+                .Should()
+                .BeTrue(executable + " should have the executable");
+        }
+
+        [Fact]
+        public void GivenNugetConfigAndPackageNameAndVersionAndTargetFrameworkWhenCallItCreateAssetFile()
+        {
+            var nugetConfigPath = WriteNugetConfigFileToPointToTheFeed();
+            var toolsPath = Path.Combine(Directory.GetCurrentDirectory(), Path.GetRandomFileName());
+
+            ExecutablePackageObtainer packageObtainer =
+                ConstructDefaultPackageObtainer(toolsPath);
+            ToolConfigurationAndExecutableDirectory toolConfigurationAndExecutableDirectory =
+                packageObtainer.ObtainAndReturnExecutablePath(
+                    packageId: TestPackageId,
+                    packageVersion: TestPackageVersion,
+                    nugetconfig: nugetConfigPath,
+                    targetframework: _testTargetframework);
+
+            var assetJsonPath = toolConfigurationAndExecutableDirectory
+                .ExecutableDirectory
+                .GetParentPath()
+                .GetParentPath()
+                .GetParentPath()
+                .GetParentPath()
+                .WithFile("project.assets.json").Value;
+
+            File.Exists(assetJsonPath)
+                .Should()
+                .BeTrue(assetJsonPath + " should be created");
+        }
+
+        [Fact]
+        public void GivenAllButNoNugetConfigFilePathtCanDownloadThePacakge()
+        {
+            var uniqueTempProjectPath = GetUniqueTempProjectPathEachTest();
+            var tempProjectDirectory = uniqueTempProjectPath.GetDirectoryPath();
+            var nugetConfigPath = WriteNugetConfigFileToPointToTheFeed();
+            var toolsPath = Path.Combine(Directory.GetCurrentDirectory(), Path.GetRandomFileName());
+
+            Directory.CreateDirectory(tempProjectDirectory.Value);
+            File.Copy(nugetConfigPath.Value,
```

Comment:
No nuget config means you don't need nuget config passed in during call

NuGet needs a way to find the package, in production, it will keep look up folders for Nuget.Config and use the feed there.

In test, we don't want NuGet to keep look up, so I just copy paste beside the project.

Reply:
Understood, thanks for the clarification!

Reply:
Added comment